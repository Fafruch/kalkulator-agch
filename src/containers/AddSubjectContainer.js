import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { addSubject } from '../actions'
import { primarySubjects } from '../constants/SubjectTypes'

let AddSubjectContainer = ({ subjectsType, onAdd }) => (
  <button
    onClick={() => onAdd(subjectsType)}
    className={subjectsType === primarySubjects ? 'btn btn-outline-danger' : 'btn btn-outline-success'}
  >
    Dodaj {subjectsType === primarySubjects ? 'przedmiot główny' : 'język'}
  </button>
)
AddSubjectContainer.propTypes = {
  subjectsType: PropTypes.string.isRequired,
  onAdd: PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  onAdd: bindActionCreators(addSubject, dispatch)
})

AddSubjectContainer = connect(null, mapDispatchToProps)(AddSubjectContainer)

export default AddSubjectContainer
