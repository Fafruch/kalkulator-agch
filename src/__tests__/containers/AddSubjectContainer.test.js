import React from 'react'
import { createStore } from 'redux'
import { shallow } from 'enzyme'

import rootReducer from '../../store/reducers'
import AddSubjectContainer from '../../containers/AddSubjectContainer'

const setup = (setupProps = {}) => {
  const defaultProps = {
    subjectsType: 'primary',
    store: createStore(rootReducer)
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<AddSubjectContainer {...props} />)

  return {
    props,
    wrapper
  }
}

describe('AddSubjectContainer component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
