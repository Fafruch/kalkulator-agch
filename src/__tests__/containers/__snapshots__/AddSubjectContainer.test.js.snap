// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSubjectContainer component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(AddSubject)
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    subjectsType="primary"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "onAdd": [Function],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subjectsType": "primary",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "onAdd": [Function],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "subjectsType": "primary",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
